### PR TITLE
cni: cilium: support cri-o container runtime engine

### PR DESCRIPTION
**Pull request by nirmoy**
## What does this PR change?

Fixes: #169

## Documentation
- No documentation needed

### CODE REVIEW

**Comment by MalloZup** [comment]
Rebase when possible

File: pkg/cni/cilium/cilium.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ const (
 
 	//CiliumCertSecret the secret name to store tls credential
 	CiliumCertSecret = "cilium-secret"
+
+	bpfMapVol = "bpf-maps"
```
Comment (MalloZup) [inline pkg/cni/cilium/cilium.go]: Add comment to bar explain usage

File: pkg/cni/cilium/cilium.go [hunk 2]
```diff
@@ -299,6 +309,36 @@ func createCiliumAddon(cniConfigMapBytes, etcdConfigMapBytes, daemonSetbytes []b
...
+		for i, volume := range ciliumDaemonSet.Spec.Template.Spec.Volumes {
+			if volume.Name == bpfMapVol {
+				index = i
+				break
+			}
+		}
+		if index != -1 {
+			// remove bpfMapVol  volume
+			ciliumDaemonSet.Spec.Template.Spec.Volumes = append(ciliumDaemonSet.Spec.Template.Spec.Volumes[:index], ciliumDaemonSet.Spec.Template.Spec.Volumes[index+1:]...)
+		}
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply pkg/cni/cilium/cilium.go]: looks better yop.

File: pkg/cni/cilium/manifests.go [hunk 3]
```diff
@@ -191,16 +192,16 @@ spec:
           initialDelaySeconds: 5
           periodSeconds: 5
         volumeMounts:
-          - name: bpf-maps
+          - name: {{ .BpfMapVol }}
```
[the remark this answers is not in the document]
Comment (nirmoy) [reply pkg/cni/cilium/manifests.go]: @inercia I did know this exist, I will use  ^^

[the remark this answers is not in the document]
Comment (inercia) [reply pkg/cni/cilium/manifests.go]: @nirmoy I've just checked and the right syntax, according to [the docs](https://golang.org/pkg/text/template/#hdr-Actions), is

```
{{if .ContainerRuntime != "crio" }} ... {{end}}
```

**Comment by MalloZup** [comment]
Thx Alvaro for time and precison

File: pkg/cni/cilium/manifests.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (inercia) [reply pkg/cni/cilium/manifests.go]: Yeah, Golang templates have their own quirks... I think it was something like `{{if ne .ContainerRuntime "crio"}}`

[the remark this answers is not in the document]
Comment (nirmoy) [reply pkg/cni/cilium/manifests.go]: tried that didn't work, using {{ if .EnableBPF }}

**Comment by nirmoy** [comment]
@MalloZup @inercia  please review again.

**Pull request merged**